### PR TITLE
M-2: Switch to using PostgreSQL instead of the In-Memory DB

### PR DESCRIPTION
**Pull request by darkness-reveur**
created migrations

### CODE REVIEW

**Review by Rickstaper — CHANGES_REQUESTED** [review]

File: MeetupPlatformApi/Program.cs [hunk 1]
```diff
@@ -9,7 +9,9 @@
 builder.Services.AddAutoMapper(typeof(Program));
 builder.Services.AddDbContext<ApplicationContext>(options =>
 {
-    options.UseInMemoryDatabase(databaseName: "MeetupPlatform");
+    options.UseNpgsql(builder.Configuration
+            .GetSection("ConnectionStrings")
```
[the remark this answers is not in the document]
Comment (darkness-reveur) [reply MeetupPlatformApi/Program.cs:13]: you are right

[the remark this answers is not in the document]
Comment (darkness-reveur) [reply MeetupPlatformApi/Program.cs]: Ready

**Review by Rickstaper — APPROVED** [review]

File: .config/dotnet-tools.json [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+  "isRoot": true,
+  "tools": {
+    "dotnet-ef": {
+      "version": "6.0.3",
+      "commands": [
+        "dotnet-ef"
+      ]
+    }
+  }
+}
```
Comment (satma0745) [inline .config/dotnet-tools.json:12]: Add empty newline at the end of the file

File: MeetupPlatformApi/Migrations/20220329130316_Initial.cs [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+    {
+        protected override void Up(MigrationBuilder migrationBuilder)
+        {
+            migrationBuilder.CreateTable(
+                name: "Meetups",
+                columns: table => new
+                {
+                    Id = table.Column<int>(type: "integer", nullable: false)
+                        .Annotation("Npgsql:ValueGenerationStrategy", NpgsqlValueGenerationStrategy.IdentityByDefaultColumn),
+                    Name = table.Column<string>(type: "text", nullable: true),
```
Comment (satma0745) [inline MeetupPlatformApi/Migrations/20220329130316_Initial.cs:19]: Why name field is nullable?

File: MeetupPlatformApi/Migrations/20220329130316_Initial.cs [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+                    StartTime = table.Column<DateTime>(type: "timestamp with time zone", nullable: false),
+                    EndTime = table.Column<DateTime>(type: "timestamp with time zone", nullable: false),
+                    Description = table.Column<string>(type: "text", nullable: true)
```
Comment (satma0745) [inline MeetupPlatformApi/Migrations/20220329130316_Initial.cs:22]: Why description field is nullable?

File: MeetupPlatformApi/Migrations/20220329130316_Initial.cs [hunk 5]
```diff
@@ -0,0 +1,36 @@
+﻿using System;
+using Microsoft.EntityFrameworkCore.Migrations;
+using Npgsql.EntityFrameworkCore.PostgreSQL.Metadata;
+
+#nullable disable
+
+namespace MeetupPlatformApi.Migrations
+{
+    public partial class Initial : Migration
```
Comment (satma0745) [inline MeetupPlatformApi/Migrations/20220329130316_Initial.cs:9]: Rename migration to "AddMeetup"

File: MeetupPlatformApi/appsettings.json [hunk 6]
```diff
@@ -4,5 +4,9 @@
       "Default": "Information",
       "Microsoft.AspNetCore": "Warning"
     }
+  },
+
+  "ConnectionStrings": {
+    "DbConnectionString": "Host=<host>;Port=<port>;Database=<db-name>;Username=<userName>;Password=<password>"
```
Comment (satma0745) [inline MeetupPlatformApi/appsettings.json:10]: Consider renaming configuration option to "PostgreSQL"

File: MeetupPlatformApi/appsettings.json [hunk 7]
```diff
@@ -4,5 +4,9 @@
       "Default": "Information",
       "Microsoft.AspNetCore": "Warning"
     }
+  },
+
```
Comment (satma0745) [inline MeetupPlatformApi/appsettings.json:8]: Consider removing redundant newline

File: MeetupPlatformApi/appsettings.json (hunk 6 above)
Comment (satma0745) [reply MeetupPlatformApi/appsettings.json:10]: Use only one convention for template naming: use either `kebab-case` like in `<db-name>` or `camelCase` like in `<userName>`

**Review by satma0745 — CHANGES_REQUESTED** [review]

File: MeetupPlatformApi/appsettings.json (hunk 7 above)
Comment (darkness-reveur) [reply MeetupPlatformApi/appsettings.json:8]: we consulted and decided to leave an empty line

File: MeetupPlatformApi/Migrations/20220329130316_AddMeetup.cs [hunk 8]
```diff
@@ -0,0 +1,36 @@
...
+        protected override void Up(MigrationBuilder migrationBuilder)
+        {
+            migrationBuilder.CreateTable(
+                name: "Meetups",
+                columns: table => new
+                {
+                    Id = table.Column<int>(type: "integer", nullable: false)
+                        .Annotation("Npgsql:ValueGenerationStrategy", NpgsqlValueGenerationStrategy.IdentityByDefaultColumn),
+                    Name = table.Column<string>(type: "text", nullable: false),
+                    StartTime = table.Column<DateTime>(type: "timestamp with time zone", nullable: false),
```
Comment (satma0745) [inline MeetupPlatformApi/Migrations/20220329130316_AddMeetup.cs:20]: Don't just change migration like this (`nullable: false`). Instead You should update Entity (add `[Required]` annotation to `Name` and `Description` field) and then re-create this migration completely from scratch. If You don't do this, we have unexpected changes in upcoming migrations. To ensure, that all migration logic is encapsulated in this migration, You can create a temporary migration, check if it's empty and remove it (this way You can ensure that last changed didn't produce any phantom affect on DB schema)

**Review by satma0745 — CHANGES_REQUESTED** [review]

File: MeetupPlatformApi/Entities/MeetupEntity.cs [hunk 9]
```diff
@@ -1,14 +1,22 @@
-﻿namespace MeetupPlatformApi.Entities;
+﻿using System.ComponentModel.DataAnnotations;
+
+namespace MeetupPlatformApi.Entities;
 
 public class MeetupEntity
 {
+    [Required]
```
Comment (satma0745) [inline MeetupPlatformApi/Entities/MeetupEntity.cs:7]: Fields with not-nullable .Net types are required by default, so there is no need to specify `[Required]` for `Id`, `Name` and `Description` fields. But, if You guys insist, we can keep them and make it our convention

File: MeetupPlatformApi/Entities/MeetupEntity.cs [hunk 10]
```diff
@@ -1,14 +1,22 @@
...
+    [Key]
```
Comment (satma0745) [inline MeetupPlatformApi/Entities/MeetupEntity.cs:8]: The same applies to the `[Key]` attribute: if field is named `Id`, then EF Core considers it a `Primary Key` by default

**Review by satma0745 — CHANGES_REQUESTED** [review]

**Pull request merged**